### PR TITLE
chore(sdk): Add dir argument for Artifact.get()

### PR DESCRIPTION
**Pull request by ramit-wandb**
Description
-----------
A user requested to be able to use `artifact.get()` to download to a specific path instead of the default `./artifacts`. This change enables that.

### <samp>🤖 Generated by Copilot at 2adc13b</samp>

Added a feature to customize the local download location of artifact files in `Artifact.get`. Modified the download logic of artifact entries and tables to support this feature.

Testing
-------
How was this PR tested?

Checklist
-------
- [x] Include reference to internal ticket "Fixes WB-NNNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [x] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/main/CONTRIBUTING.md#conventional-commits)

### <samp>🤖 Generated by Copilot at 2adc13b</samp>

> _`Artifact.get` changes_
> _Customize download root now_
> _Winter of content_

### CODE REVIEW

File: wandb/sdk/artifacts/artifact.py [hunk 1]
```diff
@@ -1480,11 +1482,11 @@ def get(self, name: str) -> Optional[data_types.WBValue]:
         # `artifact.download`. In the future, we should refactor the deserialization
```
Comment (szymon-piechowicz-wandb) [inline wandb/sdk/artifacts/artifact.py:1482]: Pass `root` on line 1475 as well.

File: wandb/sdk/artifacts/artifact.py [hunk 2]
```diff
@@ -1435,7 +1435,9 @@ def get_path(self, name: StrPath) -> ArtifactManifestEntry:
         entry._parent_artifact = self
         return entry
 
-    def get(self, name: str) -> Optional[data_types.WBValue]:
+    def get(
+        self, name: str, root: Optional[str] = None
+    ) -> Optional[data_types.WBValue]:
         """Get the WBValue object located at the artifact relative `name`.
 
         Arguments:
```
Comment (szymon-piechowicz-wandb) [inline wandb/sdk/artifacts/artifact.py:1443]: Nit: add the param to the docstring.

**Comment by szymon-piechowicz-wandb** [comment]
> chore(sdk)

Nit: this looks more like `feat(artifacts)` to me.

**Comment by cvphelps** [comment]
@ramit-wandb could you please follow up on this PR? It sounds like it would be a nice fix for Ghost https://github.com/wandb/wandb/pull/5684

**Pull request closed without merging**